worlds/factorio/Options.py

```diff
@@ -60,6 +60,18 @@ class Goal(Choice):
     default = 0
 
 
+class CraftSanity(NamedRange):
+    """Choose a number of researches to require crafting a specific item rather than with science packs.
+    Will be capped based on the total number of locations."""
+    display_name = "CraftSanity"
+    default = 0
+    range_start = 0
+    range_end = 183
+    special_range_names = {
+        "disabled": 0
+    }
+
```

Comment:
I feel like this should be a percentage, rather than hard-coded based on the current number of techs.

Reply:
If you turn on maximum traps, you can have 389 locations and there are only 183 craftable items. So, what happens when someone picks 100% CraftSanity in that case?